### DISCUSSION TITLE
Windows Prompts

### DISCUSSION THREAD

**Discussion by ncd76** [coreybutler/node-windows#340] [General]
Any way to stop several prompts after each install?

**Comment by coreybutler** [comment]
What prompts are you referring to?

**Comment by ncd76** [comment]
User Account Control - Do you want this app to make changes to your device. Most are: Net Command

**Reply by coreybutler** [reply]
UAC prompts are triggered by Windows itself anytime a sensitive operation is requested from a user without permission to perform those operations. There's no way for node-windows to prevent that, but you can run your scripts under user accounts that have appropriate permissions. For example, running Windows in developer mode can elevate some permissions automatically. Running with an admin account will prevent that.

Overall, this is all about what your script is trying to do. If you don't need to do something sensitive, don't. For example, many people try to write to `C:\Program Files`, which is a Windows protected directory. It requires elevated permissions to write to this directory. If your script can write somewhere else, that's one way to avoid the prompts. You could potentially use shims to bypass some security challenges too, but success is circumstantial. If your script really does need to do something sensitive, then the user either needs admin permissions or you have to deal with the prompts... because Windows forces that.

**Comment by ncd76** [comment]
I was just running the Hello World example

    var Service = require('node-windows').Service;
    var svc = new Service({
        name: 'Hello',
        script: require('path').join(__dirname, 'test-script.js')
    });
    svc.on('install', function () {
        svc.start();
        console.log("Started")
    });
    svc.install();

test-script.js is empty.

I'm running inside VS Code terminal. If you start VS code as Admin then the UAC controls don't appear.

Thx...

**Comment by pablofuentes88** [comment]
Same problem here,
How to avoid 4 pop up UAC autorization?
With 1 is ok but 4??!!

I know, if I execute the script `node miApp.js` in a terminal with privilegious any autorization pop-up appear.
1. Its most convenient run the script
2. Pop-up UAC appear (1 time), accept.
3. Windows Service intalled and running.

PD: Sorry for my bad english 🙈

**Comment by pablofuentes88** [comment]
I found the solution 🤣 - Autoanswer
Create 3 files, runService.js, service.js and helloworld.js

**runService.js**
```
var wincmd = require("node-windows");

wincmd.elevate('node .\\service.js')
```

**service.js**
```
var Service = require("node-windows").Service;

// Create a new service object
var svc = new Service({
  name: "Hello World",
  description: "The nodejs.org example web server.",
  script: require("path").join(__dirname, "helloworld.js"),
  nodeOptions: ["--harmony", "--max_old_space_size=4096"],
  //, workingDirectory: '...'
  //, allowServiceLogon: true
});

// Listen for the "install" event, which indicates the
// process is available as a service.
svc.on("install", function () {
  svc.start();
});

svc.install()
```

**helloworld.js**
`console.log('hola mundo')`


That's it!
